File: IOI/18/combo/java/combo.java
public class combo {
  String guess_sequence(int N) {
    String p = "";
    for (int i = 0; i < 4 * N; ++i) {
      p += 'A';
    }
    int coins = combo_lib.press(p);
    String S = "";
    for (int i = 0; i < N; ++i) {
      S += 'A';
    }
    return S;
  }
}

File: IOI/18/combo/pas/combo.pas
unit combo;

interface

function guess_sequence(N: longint): ansistring;

implementation

uses combo_lib;

function guess_sequence(N: longint): ansistring;
var
  p, S: ansistring;
  coins, i: longint;
begin
  p := '';
  for i := 1 to 4 * N do begin
    p += 'A';
  end;
  coins := press(p);
  S := '';
  for i := 1 to N do begin
    S += 'A';
  end;
  exit(S);
end;

end.
